java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -212,11 +215,20 @@ public GeneratorResult generateJava() {
         if (!prop.isNullable) {
           // from https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable :
           // "_null values for fields that either don't specify the nullable flag, or give it a false
-          // value, will be pruned before defaulting happens..._"
+          // value, will be pruned before defaulting happens. If a default is present, it will be applied_"
           objField.addAnnotation(
               new SingleMemberAnnotationExpr(
                   new Name("com.fasterxml.jackson.annotation.JsonSetter"),
                   new NameExpr("nulls = com.fasterxml.jackson.annotation.Nulls.SKIP")));
+          if (prop.defaultValue != null) {
```

Comment:
Have you attempted to move this block out from the `if` for the nullable?
If we have set all the Jackson annotations properly this might work according to spec even defining always the body block.

Reply:
Right, fixed.